---
title: C26111 | Microsoft Docs
ms.custom: ''
ms.date: 2018-06-30
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
caps.latest.revision: 10
author: corob-msft
ms.author: gewarren
manager: ghogen
ms.openlocfilehash: 22e2b3663c24d13ae92d1d8aa7a34a2dd513bba4
ms.sourcegitcommit: 55f7ce2d5d2e458e35c45787f1935b237ee5c9f8
ms.translationtype: MT
ms.contentlocale: es-ES
ms.lasthandoff: 08/22/2018
ms.locfileid: "47582811"
---
# <a name="c26111"></a>C26111
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

La versión más reciente de este tema puede encontrarse en [C26111](https://docs.microsoft.com/visualstudio/code-quality/c26111).  
  
Advertencia C26111: error del llamador al desbloquear \<bloqueo > antes de llamar a función \<func >.  
  
 La anotación `_Requires_lock_not_held_` impone una condición previa que el recuento de bloqueos del bloqueo especificado no puede ser mayor que cero cuando se llama a la función. C26111 de advertencia se emite cuando se produce un error en una función liberar el bloqueo antes de llamar a otra función.  
  
## <a name="example"></a>Ejemplo  
 En el ejemplo siguiente genera la advertencia C26111 porque la `_Requires_lock_not_held_` se infringe la condición previa al llamar a `DoNotLock` dentro de la sección bloqueada.  
  
```  
  
typedef struct _DATA   
{  
    CRITICAL_SECTION cs;  
    int d;  
} DATA;  
  
_Requires_lock_not_held_(p->cs)   
  
void DoNotLock(DATA* p)  
{  
    EnterCriticalSection(&p->cs);   
    p->d = 0;  
    LeaveCriticalSection(&p->cs);  
}  
  
void LockedFunction(DATA* p)   
{   
    EnterCriticalSection(&p->cs);   
    DoNotLock(p); // Warning C26111  
    LeaveCriticalSection(&p->cs);  
}  
  
```



